cmd/orphan-finder/main.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/codegangsta/cli"
+
+	"github.com/letsencrypt/boulder/cmd"
+	"github.com/letsencrypt/boulder/core"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/rpc"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+type config struct {
+	AMQP   cmd.AMQPConfig
+	Statsd cmd.StatsdConfig
+	Syslog cmd.SyslogConfig
+}
+
+var (
+	b64derOrphan = regexp.MustCompile(`b64der=\[([a-zA-Z0-9+/]+)\]`)
+	regOrphan    = regexp.MustCompile(`regID=\[(\d+)\]`)
+)
+
+func checkDER(sai core.StorageAuthority, der []byte) error {
+	cert, err := x509.ParseCertificate(der)
+	if err != nil {
+		return fmt.Errorf("Failed to parse DER: %s", err)
+	}
+	_, err = sai.GetCertificate(core.SerialToString(cert.SerialNumber))
+	if err != nil && err != sa.ErrCertNotFound {
```

Comment:
Ack, I realized this comparison is not actually going to work because this happens across RPC boundaries and `error` is a reference type, so this will be checking reference equality and will fail. There's a `core.NotFoundError` which is an error struct and is marshalled/unmarshalled by the RPC layer, I think that should work.